scripts/update-browser-releases/edge.ts

```diff
@@ -0,0 +1,197 @@
+/* This file is a part of @mdn/browser-compat-data
+ * See LICENSE file for more information. */
+
+import * as fs from 'node:fs';
+
+import chalk from 'chalk-template';
+
+import { newBrowserEntry, updateBrowserEntry } from './utils.js';
+
+/**
+ * getReleaseNotesURL - Guess the URL of the release notes
+ * @param {string} status The status of the release
+ * @param {string} fullVersion The version of the release
+ * @param {string} date The date of the release
+ * @returns {string} The URL of the release notes or the empty string if not found
+ */
+const getReleaseNotesURL = async (status, fullVersion, date) => {
+  // If the status isn't stable, do not give back any release notes.
+  if (status !== 'Stable') {
+    return '';
+  }
+  console.log(fullVersion);
+  const versionStr = fullVersion.replace(/\./g, '');
+  const month = [
+    'january',
+    'february',
+    'march',
+    'april',
+    'may',
+    'june',
+    'july',
+    'august',
+    'september',
+    'october',
+    'november',
+    'december',
+  ];
+  const dateObj = new Date(date);
+  const dateStr = `${
+    month[dateObj.getMonth()]
+  }-${dateObj.getDate()}-${dateObj.getFullYear()}`;
+  return `https://learn.microsoft.com/en-us/deployedge/microsoft-edge-relnote-stable-channel#version-${versionStr}-${dateStr}`;
+};
+
+/**
+ * updateEdgeReleases - Update the json file listing the browser releases of an Edge browser
+ * @param {object} options The list of options for Edge.
+ */
+export const updateEdgeReleases = async (options) => {
+  //
+  // Get the JSON with the versions from chromestatus
+  //
+  const buffer = await fetch(options.edgeupdatesURL);
+  const edgeVersions = JSON.parse(await buffer.text());
+
+  //
+  // Get the chrome.json from the local BCD
+  //
+  const file = fs.readFileSync(`${options.bcdFile}`);
+  const edgeBCD = JSON.parse(file.toString());
+
+  //
+  // Update the three channels
+  //
+  const channels = new Map([
+    ['current', options.releaseBranch],
+    ['beta', options.betaBranch],
+    ['nightly', options.nightlyBranch],
+  ]);
+
+  // Information store to be used outside the for loop
+  const data = {};
+
+  for (const [key, value] of channels) {
+    //
+    // Extract the useful data
+    //
+
+    // Find the right entry
+    const entry = edgeVersions.find((e) => e['Product'] === value);
+
+    // Get the last version
+    const version = parseFloat(entry['Releases'][0]['ProductVersion']);
+
+    // We need to find the first release with this version:
```

Comment:
Maybe explain that releases are found in the JSON from newest to oldest, and so we need to go and get the first one to get the date when version XYZ was first released to the world (subsequent releases being patches only).

Reply:
Fixed.